pandas/_libs/algos.pyx

```diff
@@ -797,212 +797,202 @@ def rank_1d(
     bint ascending=True,
     na_option="keep",
     bint pct=False,
+    labels=None,
```

Comment:
type them as: `const int64_t[:]` note you may have to *always* pass grouplabels (you can just pass all 0's for a non-grouped rank). it makes these functions much simpler. and put the args in the same order, e.g. e.g. labels should be right after in_arr.

genreally I am +1 on combining grouped / non-grouped operations. currently we often have 2 (or more) versions of these. @mroeschke is working on something akin to this as well (orthogonal to this PR).

Reply:
Makes sense. Original refactor looked more like that, but changed to this because ran into annoying issues where `rank_1d` is called from other algos in multiple places (`nancorr_spearman`, `rank_2d`) and I wanted to avoid modifying those. But having labels as `const int64_t[:]` definitely makes the logic cleaner, will change back

Reply:
Done